### DISCUSSION TITLE
Display File type in the for file fields in Directus 9 (node)

### DISCUSSION THREAD

**Discussion by Pilskalns** [directus/directus#3616] [Feature Requests]
Here's a little problem I am facing when getting a collection item:

![image](https://user-images.githubusercontent.com/6931156/104012879-426a6280-51b9-11eb-8c43-380152950050.png)

The File (from the File Library) with field name `image` looks the "same" as UUIDs for user fields. (_yes, values are different, but for code consuming the API_).

If I check the `/fields/news/image`, the type is a string:
![image](https://user-images.githubusercontent.com/6931156/104013079-9ecd8200-51b9-11eb-80ab-973dffb13331.png)

The workaround I am using is to try download from `/assets/{uuid}` everything that looks like UUID and just reject errors.
Even more so, non-existent items return 403 Unauthorized. Wouldn't the 404 be more suited for it? (Even if the user with such UUID exist, I tried to get an asset with such ID, and that one doesn't exist).

I dunno if this is a bug or improvement request, hence creating a discussion here.

**Comment by Pilskalns** [comment]
Ok, I kinda figured it out... the `/fields` returns `data.meta.interface = image`. This is how the user interacted with the field and was saved as a string. So completely abstracted in how it is stored, how UI edited and how it is served through API.

**Comment by rijkvanzanten** [comment]
I might be a little confused as to how you're reading values from the API 🤔 Seeing that you are in control of the data model, wouldn't the fact that the field is named `image` already be enough of a pointer to know it's an image?

**Comment by Pilskalns** [comment]
The screenshots were from the browser, but using Javascript SDK for the code.

I am writing a little middle layer that translates content from the API to markdown files for the Hugo static site. And wanted to make things as transparent between both ends as possible. I.e. if I add a field `portrait` to CMS, then such field and a file would end up in the frontmatter for Hugo without changing the middle layer.

**Comment by Pilskalns** [comment]
`data.meta.interface` is really what I need - it can be also used to tell apart different kinds of text display types. I.e. WYSIWYG vs Markdown field type.

I think this thread can be closed. Thanks for the comment!

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
